fnConsensus/reactor.go

```diff
@@ -475,7 +455,7 @@ func (f *FnConsensusReactor) vote(fnID string, fn Fn, currentValidators *types.V
 	if aggregateExecutionResponse != nil {
 		f.safeSubmitMultiSignedMessage(
 			fn,
-			safeCopyBytes(aggregateExecutionResponse.Hash),
+			safeCopyBytes(f.state.Message[fnID]),
```

Comment:
`stateMtx` is required to access hash map in state to prevent race condition. Also need to make sure that acquiring the lock will not cause deadlock

Reply:
Mutex is already acquired at this point, so this looks fine.